.circleci/packages/config.yaml

```diff
@@ -0,0 +1,51 @@
+
+version:
+  release:
+    match: '^v[0-9]+.[0-9]+.[0-9]+'
+    value: '{{env.CIRCLE_TAG[1:]}}'
+  default:
+    value: '3.x-{{env.CIRCLE_SHA1[:8]}}'
```

Comment:
This default value is difficult to work with. Consider:
```
$ dpkg --compare-versions 3.0.0 lt 3.0.1 && echo yes  # good
yes

$ dpkg --compare-versions 3.0.0 lt 3.x-0123abcd && echo yes  # maybe not good?
yes

$ dpkg --compare-versions 3.x-gfed9876 lt 3.x-0123abcd && echo yes  # but what if 3.x-0123abcd was created before 3.x-gfed9876?
[1]
```

Part of the reason why I'm commenting on this is that we are uploading to `influxdb/releases/influxdb3`, but these aren't releases, they're snapshots. Package managers are going to have trouble dealing with these versions. I would advocate that the `influxdb/releases/influxdb3` only contain things that are considered a release and use something else for snapshots or nightlies.

Fyi, `telegraf` has `telegraf/nightlies/telegraf_nightly_amd64.deb`, which gets overwritten every night. Maybe that isn't what you want right now though (ie, perhaps it is useful to have different snapshots). 

As a strawperson, I suggest that:
1. someone creates `influxdb/snapshots/influxdb3` and move everything that was uploaded to `influxdb/releases/influxdb3` there. New snapshots go to `influxdb/snapshots/influxdb3` and use `3.0.0+snapshot-{{env.CIRCLE_SHA1[:8]}}` as the default.
2. if one of the current uploads was an mvp, tag it with `3.0.0-0.alpha.0.0` and put in `influxdb/releases/influxdb3`. For important internal pre-alpha releases, you can tag with `3.0.0-0.alpha.0.1`, `3.0.0-0.alpha.0.2`, etc. When official alpha 1 is released, tag with `3.0.0-0.alpha.1`, then use `3.0.0-0.alpha.2` for 2nd official alpha, and so on. Continue with `3.0.0-0.beta.1`, `3.0.0-0.rc.1` as desired. This versioning scheme works with both semver and packaging. Ie: `3.0.0-0.alpha.0.0` < `3.0.0-0.alpha.0.1` < `3.0.0-0.alpha.0.2` < `3.0.0-0.alpha.1` < `3.0.0-0.alpha.2` < `3.0.0-0.beta.1` < `3.0.0-0.rc.1`. All of these tagged releases go to `influxdb/releases/influxdb3`
3. when 3.0.0 is released, tag as `3.0.0`. semver is happy since `3.0.0-0.rc.1` < `3.0.0` and packaging is updated to use `3.0.0-1` such that `3.0.0-0.rc.1` < `3.0.0-1`

In the above, we don't worry if `3.0.0+snapshot-{{env.CIRCLE_SHA1[:8]}}` is higher or lower in packaging since people will be installing those manually.

Reply:
This is _definitely_ a good point. I may go ahead and update these for 1.X OSS/Enterprise and 2.X OSS. As you alluded to, the default version schemes weren't intended to be consumed by the general public. So I had just written them to make `apt` and `yum` happy. However, I definitely can foresee a future where someone makes these available to the general public (not here, but in 1.X and 2.X land).